explore/marginal-gain/1.Marginal-Gain-TP53-Example.py

```diff
@@ -0,0 +1,327 @@
+
+# In[167]:
+
+import os
+import urllib
+import random
+import warnings
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn import preprocessing
+from sklearn.linear_model import SGDClassifier
+from sklearn.model_selection import train_test_split, GridSearchCV
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler, Imputer
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+from IPython.display import display
+import gc
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[3]:
+
+# We're going to be building a 'TP53' classifier 
+GENE = '7157' # TP53
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+# 
+# *Here is some [information](https://ghr.nlm.nih.gov/gene/TP53) about TP53*
+
+# ## Load Data
+
+# In[4]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..', '..', 'download', 'covariates.tsv')\ncovariates = pd.read_table(path, index_col=0)\npath = os.path.join('..', '..', 'download', 'expression-matrix.tsv.bz2')\nexpression = pd.read_table(path, index_col=0)")
+
+
+# In[14]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..','..','download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[278]:
+
+X = {}
+X['a'] = covariates
+X['b'] = pd.concat([covariates,expression], axis=1)
+for k in ['a','b']:
+    print(X[k].shape)
+
+
+# In[147]:
+
+# Which columns have NaN values? 
+for k in ['a','b']:
+    print(X[k].columns[pd.isnull(X[k]).any()].tolist())
+
+
+# In[159]:
+
+y = Y[GENE]
+
+
+# In[160]:
+
+# The Series now holds TP53 Mutation Status for each Sample
+y.head(6)
+
+
+# In[164]:
+
+# Here are the percentage of tumors with NF1
+y.value_counts(True)
+
+
+# ## Set aside 10% of the data for testing
+
+# In[165]:
+
+X_train = {}
+X_test = {}
+for k in ['a','b']:
+    X_train[k], X_test[k], y_train, y_test = train_test_split(X[k], y, test_size=0.1, random_state=0)
+    'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X[k].columns), 
+                                                                              len(X_train[k]), len(X_test[k]))
+
+
+# ## Median absolute deviation feature selection
+
+# In[20]:
+
+def fs_mad(x, y):
+    """    
+    Get the median absolute deviation (MAD) for each column of x
+    """
+    scores = mad(x) 
+    return scores, np.array([np.NaN]*len(scores))
+
+
+# ## Define pipeline and Cross validation model fitting
+
+# In[22]:
+
+# Parameter Sweep for Hyperparameters
+param_grid = {
+    'classify__loss': ['log'],
+    'classify__penalty': ['elasticnet'],
+    'classify__alpha': [10 ** x for x in range(-3, 1)],
```

Comment:
Looks like your notebook only has 1 value here. Reminder to reset when you're ready for a final analysis. I think it's also fine to consolidate to a single notebook later (probably the current `2.`).